### ISSUE TITLE
Pickle error on deferred fields as of 0.8.5

### ISSUE THREAD

**Issue by vdboor** [django-mptt/django-mptt#494]
In 0.8.5 there is an error with deferred fields. I've been able to bisect it to b0c8561876011e0cdd561a4fec5c0c4c0551b4af

Example model:

```
from django.db import models
from mptt.models import MPTTModel, TreeForeignKey


class Foo(MPTTModel):
    parent = TreeForeignKey('self', null=True)
    title1 = models.CharField(max_length=10)
    title2 = models.CharField(max_length=10)
```

Test case:
```
from django.test import TestCase
from .models import Foo 
import pickle


class FooTestCase(TestCase):
    def test_foo(self):
        Foo.objects.create(title1="11", title2="22")

        qs = Foo.objects.only('title1')
        pickle.dumps(list(qs), protocol=pickle.HIGHEST_PROTOCOL)
```

This happens in one of my projects where a deferred queryset (having a summary of the table) is cached in memcached.

When the following line is removed from the `MPTTModel.__init__()` method, the pickling above works:

`self._tree_manager = self._tree_manager.db_manager(self._state.db)`

I hope this greatly helps finding this bug. My mptt knowledge kind of stops at that point..

**Comment by vdboor** [comment]
It seems #484 touches the same issue, this test case could serve for debugging it.

**Comment by craigds** [comment]
thanks. merged fix (see #484)
